src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -19,9 +22,30 @@ async def run(*actors: Any) -> None:
     Raises:
         AssertionError: if any of the actors is not an instance of BaseActor.
     """
+    pending_tasks = set()
     # Check that each actor is an instance of BaseActor at runtime,
     # due to the indirection created by the actor decorator.
     for actor in actors:
         assert isinstance(actor, BaseActor), f"{actor} is not an instance of BaseActor"
-
-    await asyncio.gather(*(actor.join() for actor in actors))
+        pending_tasks.add(asyncio.create_task(actor.join(), name=str(actor)))  # type: ignore
```

Comment:
You should stop created tasks before raising assert error.

Reply:
This looks more resilient to me :+1:

Reply:
Then I'll update it like that

Reply:
Looks good:)